### PR TITLE
Added make Gurkan

### LINKED ISSUE #32: Automatically give Gurkan role on name change
This feature would have the bot check a member's name every time it changes and if a variant of gurkan is in the name, it would give the gurkan role, else if it is not, it would remove the role. This would be done using the "on_member_update" function of discord.py.

### PR DESCRIPTION
**Pull request by RandomDev26**
Closes #32

### CODE REVIEW

**Comment by RandomDev26** [comment]
crap, forgot to lint, I'll do it rn

File: bot/exts/utils/makegurkan.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+from discord.ext import commands
+from discord.utils import find
+import re
+
+
+class MakeGurkan(commands.Cog):
+    def __init__(self, bot):
```
Comment (Shivansh-007) [inline bot/exts/utils/makegurkan.py:7]:
```suggestion
    def __init__(self, bot: Bot) -> None:
```
Just remember to import it:
```py
from bot.bot import Bot
```

File: bot/exts/utils/makegurkan.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        self.bot = bot
+
+    @commands.Cog.listener()
+    async def on_member_update(self, before, after):
```
Comment (Shivansh-007) [inline bot/exts/utils/makegurkan.py:11]:
```suggestion
    async def on_member_update(self, before: discord.Member, after: discord.Member) -> None:
        """When the member updates their nicknames, this function will check if member is classified to be a gurkan or not."""
```
Remember to import `discord`

File: bot/exts/utils/makegurkan.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+            return
+        if before.display_name != after.display_name:
+            role = find(lambda r: r.name == "Gurkans", after.guild.roles)
+            if re.search(r"g?urk?a?n?", str(after.display_name)):
+                await after.add_roles(role)
+            else:
+                await after.remove_roles(role)
+
+    @commands.Cog.listener()
+    async def on_member_join(self, member):
```
Comment (Shivansh-007) [inline bot/exts/utils/makegurkan.py:22]:
```suggestion
    async def on_member_join(self, member: discord.Member) -> None:
         """When the member updates their username, this function will check if member is classified to be a gurkan or not."""
```

File: bot/exts/utils/makegurkan.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
+from discord.ext import commands
+from discord.utils import find
+import re
```
Comment (Shivansh-007) [inline bot/exts/utils/makegurkan.py:3]:
```suggestion
import re

import discord
from bot.bot import Bot
from discord.ext import commands
from discord.utils import find
```

File: bot/exts/utils/makegurkan.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+class MakeGurkan(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.Cog.listener()
+    async def on_member_update(self, before, after):
+        if after.bot:
+            return
+        if before.display_name != after.display_name:
+            role = find(lambda r: r.name == "Gurkans", after.guild.roles)
```
Comment (Shivansh-007) [inline bot/exts/utils/makegurkan.py:15]: Instead of name use the id of the role, which should be present in the `constants.py` file

**Review by Shivansh-007 — CHANGES_REQUESTED** [review]

**Review by gustavwilliam — CHANGES_REQUESTED** [review]

File: bot/exts/utils/makegurkan.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+            else:
+                await after.remove_roles(role)
+
+    @commands.Cog.listener()
+    async def on_member_join(self, member: discord.Member) -> None:
+         """When the member updates their username, this function will check if member is classified to be a gurkan or not."""
+        if member.bot:
+            return
+        role = find(lambda r: r.name == "Gurkans", member.guild.roles)
+        if re.search(r"g?urk?a?n?", str(member.name)):
```
[the remark this answers is not in the document]
Comment (gustavwilliam) [reply bot/exts/utils/makegurkan.py:31]: Actually, I think that function is merged now, meaning you can use it.

**Comment by Shivansh-007** [comment]
@RandomDev26 Are you still working on this?

**Comment by gustavwilliam** [comment]
@RandomDev26 whenever you have time, I'd like to see this merged soon. Don't work on other issues until you've finished this one.

**Comment by RandomDev26** [comment]
:thumbsup:

**Comment by RandomDev26** [comment]
I need to close this, for some reason my new commits aren't showing, I'll make a new PR

**Pull request closed without merging**